mlworkflow/datasets.py

```diff
@@ -117,6 +117,21 @@ def _from_dict(obj, dic):
         return from_dict(dic)
     return type(obj)(**dic)
 
+# transforms a list of (key, value) items (dictionaries or dictionarizable objects)
+# into a dictionary of (key, wrapped values)
+def batchify(items, wrapper=np.array):
+    values = {}
+    for item in items:
+        item = _to_dict(item)
+        if not values:
+            values = {k:[v] for k, v in item.items()}
+        else:
+            for k in values:
+                values[k].append(item[k])
+    for k in values:
+        values[k] = wrapper(values[k])
+    return items
```

Comment:
So at this point, `items` may actually have varying sizes, I am wondering whether it is a good idea given that batch_norm, ... don't deal well with batch size of 1 for instance. It may even be empty while it is not often the case that the code behind will handle that case.

I like the idea of being able to skip certain items for some reason, but I think we should do it lazily before building the batches, so we cannot split the dataset keys into batches directly.

Reply:
no, no, items always have the same size (and this is handled by chunkify). 
Chunkify has the responsibility to return chunks of equal sizes (I would btw put the drop_last argument to True by default).

This piece of code is just the factorization of the query method. No behavior has change